### DISCUSSION TITLE
Works great locally but how do I access it from remote?

### DISCUSSION THREAD

**Discussion by spantobi** [drakkan/sftpgo#561] [Q&A] [answered]
Hi!

Sorry for the newbie questions but I managed to get sftpgo working locally in Debian 10. I created a user in the web admin, I was able to add a shared folder from my NAS and I can connect to the server with my local IP. But where do I go from here if I want to be able to access the server from remote?

I changed the "port": under "sftpd" to 1910 (there's 2 sftpd ports in the sftpgo.jso. I changed both because I don't know the difference between them)

I opened port 1910, 50000-50100 in PFSense and pointed them to my local sftpgo machine's IP.

I confirmed that port 1910 is OPEN.

Here are my Pfsense settings in /Firewall/NAT/Port Forward:

Port 1910:

Destination: WAN address
Destination Port range: Other, 1910, Other, 1910
Redirect Target IP: Single Host, "sftp local ip"
Redirect Target Port: Other, 1910

Port 50000-50100:

Destination: WAN address
Destination Port range: Other, 50000, Other, 50100
Redirect Target IP: Single Host, "sftp local ip"
Redirect Target Port: Other, 50000-50100

What else do I need to do, is there a guide for people like me? I tried connecting to it but it times out so I guess I'm just going about this the completely wrong way :)

Thanks!

**Comment by drakkan** [comment, marked as answer]
Hi,

the default sftpgo.json configuration file has just one SFTP port:

```
 "sftpd": {
    "bindings": [
      {
        "port": 2022,
        "address": "",
        "apply_proxy_config": true
      }
    ],
```

the default is 2022. If you change the port you have to restart the SFTPGo service to apply the change. 

The port range `50000-50100` is for passive FTP. If you are only using SFTP you just need to open the configured SFTP port. Please also make sure that your debian server use your PFSense as default gateway

**Comment by spantobi** [comment]
Hi!

Thank you for the quick reply!

So after a few hours of trying different things, I finally figured it out.

I changed the port back to the default 2022, here:

>   "sftpd": {
    "bindings": [
      {
        "port": 2022,
        "address": "",
        "apply_proxy_config": true
> 

and changed the port back to zero, here:

>"ftpd": {
    "bindings": [
      {
        "port": 0,
        "address": "",
        "apply_proxy_config": true,
        "tls_mode": 0,
        "force_passive_ip": "",
        "client_auth_type": 0,
        "tls_cipher_suites": [],
        "passive_connections_security": 0,
        "active_connections_security": 0,
        "debug": false
>

and restarted sftpgo:

I then removed the 50000-50100 and 1910 port forwarding in Pfsense and then added a new port forwarding to 2022 like this:

Destination: WAN address
Destination Port range: Other, 2022, Other, 2022
Redirect Target IP: Single Host, "The local IP of the sftpgo server"
Redirect Target Port: Other, 2022
NAT reflection: Enable (NAT + Proxy)

The NAT reflection seems to be the thing that made it work.

I can now connect to it using Windows Powershell using "sftp -P 2022 user@myWANipAddress"
and I get a ECDSA keyprint etc.

I really appreciate your help, thanks again! :)

**Reply by drakkan** [reply]
Thanks for sharing!

If you set the same port for both SFTP and FTP service SFTPGo will not start at all, if you also want FTP service o WebDAV service you have to set different ports
